### PR TITLE
Check block signatures before node start

### LINKED ISSUE #509: Verify signatures failed when node start
The following error might occur when node upgrade from 0.18.0 to 0.19.0+,
```
[skycoin.main:ERROR] Invalid block signatures: Verify signature of block in depth: 
2708 failed: no sig for <SIG>
```
To fix this issue, we need to parse the block again to fill the lost signature when node start.

### PR DESCRIPTION
**Pull request by iketheadore**
Remove the data.db and pull blocks again if found  signature lost.

This should be able to fix #509.

### CODE REVIEW

File: src/visor/blockchain.go [hunk 1]
```diff
@@ -615,36 +675,77 @@ func (bc Blockchain) TransactionFee(t *coin.Transaction) (uint64, error) {
...
+	for i := 0; i < SigVerifyTheadNum; i++ {
+		wg.Add(1)
+		go func(id int) {
+			defer wg.Done()
+			for {
+				select {
+				case <-c:
+					return
+				case seq := <-seqC:
+					b := bc.GetBlockInDepth(seq)
```
Comment (gz-c) [inline src/visor/blockchain.go]: Leave this as a separate function that can be call to check a block seq

File: src/visor/visor.go [hunk 2]
```diff
@@ -157,19 +157,32 @@ func NewVisor(c Config) (*Visor, VsClose, error) {
...
+			return nil, nil, err
+		}
+		logger.Error("signature lost...")
+		closeDB()
 
-	// creates block signature bucket
-	sigs, err := blockdb.NewBlockSigs(db)
-	if err != nil {
-		return nil, nil, err
+		// remove the db
```
Comment (gz-c) [inline src/visor/visor.go]: can you make this logic a separate function

File: src/visor/blockchain.go [hunk 3]
```diff
@@ -337,6 +370,27 @@ func (bc *Blockchain) ExecuteBlock(tx *bolt.Tx, b *coin.Block) error {
 	return nil
 }
 
+// ExecuteBlock attempts to append block to blockchain
+func (bc *Blockchain) ExecuteBlock(sb *coin.SignedBlock) error {
```
Comment (gz-c) [inline src/visor/blockchain.go]: If only test uses this, remove it, have test use bolt.Tx

File: src/visor/unconfirmed.go [hunk 4]
```diff
@@ -339,28 +338,30 @@ func (utp *UnconfirmedTxnPool) InjectTxn(bc *Blockchain, t coin.Transaction) (bo
...
 		now := utc.Now().UnixNano()
 		tx.Received = now
 		tx.Checked = now
-
-		if unspentNotExistErr == nil {
-			tx.IsValid = 1
-		} else {
-			tx.IsValid = 0
-		}
+		tx.IsValid = 1
```
Comment (gz-c) [inline src/visor/unconfirmed.go]: We don't save unconfirmedtxns missing unspents now?

File: src/visor/visor.go [hunk 5]
```diff
@@ -198,13 +182,48 @@ func NewVisor(c Config) (*Visor, VsClose, error) {
...
+// the db and create an empty blockchain.
+func load(dbpath string, pubkey cipher.PubKey, arbitrating bool) (*bolt.DB, *Blockchain, error) {
+	// creates blockchain instance
+	db, err := openDB(dbpath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	bc, err := NewBlockchain(db, pubkey, Arbitrating(arbitrating))
+	if err != nil {
```
Comment (gz-c) [inline src/visor/visor.go]: This should only remove db if the error type is invalid db type error

File: src/visor/unconfirmed.go (hunk 4 above)
Comment (iketheadore) [reply src/visor/unconfirmed.go]: 1. Yes, we don't save the tx that missing unspents now, actually we will check unconfirmed txs and remove those missing unspents txns when node start. 
2. VerifyTransactionFee method won't return ErrUnspentNotExist, and no one returns `ErrUnspentNotExist`.

File: src/visor/visor.go [hunk 6]
```diff
@@ -198,13 +182,52 @@ func NewVisor(c Config) (*Visor, VsClose, error) {
...
+		if err != ErrSignatureLost {
+			return nil, nil, err
+		}
+
+		logger.Error("%v", err)
+		db.Close()
+
```
Comment (gz-c) [inline src/visor/visor.go]: Log message should say that it is removing database because invalid signatures

**Pull request merged**